src/peer_manager.rs

```diff
@@ -734,6 +734,26 @@ impl PeerManager {
             .map(|(name, _)| *name);
         for name in removal_keys.iter() {
             let _ = self.candidates.remove(name);
+            let _ = self.peer_map.remove_by_name(&name).and_then(|peer| match peer {
```

Comment:
Couldn't this whole block be avoided if we just chain `removal_keys` with `names_to_drop.into_iter()` at line 699 and add the `PeerState::Candidate(RoutingConnection::JoiningNode(timestamp))` variant alongside the similar `PeerState::Routing` one?  Also, that would mean the removed candidate's Peer ID gets added to `ids_to_drop`, which we should do in any case.